qiskit/pulse/ops.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Schedule operations.
+"""
+import logging
+from typing import List
+
+from .interfaces import ScheduleComponent
+from .schedule import Schedule
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=missing-return-doc,missing-type-doc
+
+
+def union(*schedules: List[ScheduleComponent],
+          shift_time: int = 0, name: str = None) -> Schedule:
+    """Create a union (and also shift if desired) of all input `Schedule`s.
+
+    Args:
+        *schedules: Schedules to take the union of
+        shift_time: Time to shift by
+        name: Name of the new schedule. Defaults to first element of `schedules`
+    """
+    if name is None and schedules:
+        name = schedules[0].name
+    return Schedule(*schedules, shift_time=shift_time, name=name)
+
+# pylint: enable=missing-type-doc
+
+
+def shift(schedule: ScheduleComponent, time: int, name: str = None) -> Schedule:
+    """Return schedule shifted by `time`.
+
+    Args:
+        schedule: The schedule to shift
+        time: The time to shift by
+        name: Name of shifted schedule. Defaults to name of `schedule`
+    """
+    if name is None:
+        name = schedule.name
+    return Schedule(schedule, shift_time=time, name=name)
+
+
+def insert(parent: ScheduleComponent, start_time: int, child: ScheduleComponent,
```

Comment:
How about rename variables `parent` and `child` to `base` and `additive` in `ops.append()` and `ops.insert()`? (Please tweak them better English words) I think they have no parent-child relation anymore, and new names align more with what append/insert actually do.